util/test/Test/Pos/Util/Golden.hs

```diff
@@ -35,6 +37,27 @@ embedGoldenTest :: FilePath -> ExpQ
 embedGoldenTest path =
     makeRelativeToTestDir ("golden/" <> path) >>= embedStringFile
 
+-- | Test if prettified JSON and unformatted JSON are equivalent.
+goldenFileEquiv :: (Eq a)
+                => Either String a -> Either String a -> Either String Bool
+goldenFileEquiv prettified unformatted = do
+        case prettified of
+            Left err -> Left $ "could not decode: " <> show err
+            Right x ->  case unformatted of
+                            Left err -> Left $ "could not decode: " <> show err
+                            Right x' -> Right $ x' == x
+
+-- | Test if prettified canonical JSON and unformatted canonical
+-- JSON are equivalent.
+goldenFileCanonicalEquiv :: HasCallStack => FilePath -> FilePath -> Property
+goldenFileCanonicalEquiv pPath uPath = withFrozenCallStack $ do
+    withTests 1 . property $ do
+        pStr <- liftIO $ readFile pPath
+        uBs <- liftIO $ LB.readFile uPath
+        case Canonical.parseCanonicalJSON uBs of
+            Left err ->  failWith Nothing $ "could not decode: " <> show err
+            Right jsVal -> (toText $ Canonical.prettyCanonicalJSON jsVal) === pStr
```

Comment:
👍 

Note to self: this function reads both files, decodes the unformatted bytestring, pretty-prints it, and asserts that the pretty-file bytestring is equal to the unformatted-then-prettyprinted bytestring.